# Gold Price Analysis Project
## Data Source & Attribution

This analysis is based on the **Historical Gold Prices (1995–2026)** provided by [Kaggle](https://www.kaggle.com/).

* **Dataset Source:** https://www.kaggle.com/datasets/mr1rameez/historical-gold-prices-19952026
* **License:** [Creative Commons Attribution 4.0 International (CC BY 4.0)](https://creativecommons.org/licenses/by/4.0/)
* **Author/Publisher:** Muhammad Rameez

> **Notice:** As required by the CC BY 4.0 license, any modifications, data cleaning, or transformations performed on the original dataset are documented in the code sections below.

In [1]:
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
#Load the historical gold price dataset (Aug 2000 - Feb 2026) using Pandas
df = pd.read_csv('Dataset/gold_prices_aug_2000_2026_feb.csv')

In [3]:
df.head(10)

Date  Gold_Price_USD_YFinance
0  2000-08-01               276.099991
1  2000-09-01               273.389996
2  2000-10-01               269.809090
3  2000-11-01               265.874997
4  2000-12-01               271.515005
5  2001-01-01               265.371427
6  2001-02-01               261.805263
7  2001-03-01               262.290907
8  2001-04-01               261.079997
9  2001-05-01               272.077272

In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 307 entries, 0 to 306
Data columns (total 2 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Date                     307 non-null    str    
 1   Gold_Price_USD_YFinance  307 non-null    float64
dtypes: float64(1), str(1)
memory usage: 4.9 KB


Using df.info() gives the following information about the DataFrame: 

The dataset contains 307 rows and 2 columns. The columns are:

Date – 307 non-null entries, type str (dates stored as text)

Gold_Price_USD_YFinance – 307 non-null entries, type float64 (gold prices in USD)

The DataFrame uses approximately 4.9 KB of memory.

There are no missing values, so the dataset is complete. You may consider converting the Date column to datetime for easier time-series analysis.

Note: The Date column is currently stored as strings. To perform time-series analysis, it is recommended to convert it to datetime format using pd.to_datetime().

In [6]:
df['Date'] = pd.to_datetime(df['Date'])

In [7]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 307 entries, 0 to 306
Data columns (total 2 columns):
 #   Column                   Non-Null Count  Dtype         
---  ------                   --------------  -----         
 0   Date                     307 non-null    datetime64[us]
 1   Gold_Price_USD_YFinance  307 non-null    float64       
dtypes: datetime64[us](1), float64(1)
memory usage: 4.9 KB
